mbtools/generate_quiz_csv.py

```diff
@@ -0,0 +1,125 @@
+import argparse
+import os
+import csv
+from mbtools import utils
+from pathlib import Path
+from mbtools import models
+
+
+def write_quiz_csvs(
+    quiz_questions_csv,
+    quiz_question_contents_csv,
+    quiz_multichoice_answers_csv,
+    output_path
+):
+    with open(output_path / "quiz_questions.csv", "w") as outfile:
+        headers = quiz_questions_csv[0].keys()
+        result = csv.DictWriter(outfile, fieldnames=headers)
+        result.writeheader()
+        result.writerows(quiz_questions_csv)
+
+    with open(output_path / "quiz_question_contents.csv", "w") as outfile:
+        headers = quiz_question_contents_csv[0].keys()
+        result = csv.DictWriter(outfile, fieldnames=headers)
+        result.writeheader()
+        result.writerows(quiz_question_contents_csv)
+
+    with open(output_path / "quiz_multichoice_answers.csv", "w") as outfile:
+        if len(quiz_multichoice_answers_csv) != 0:
+            headers = quiz_multichoice_answers_csv[0].keys()
+        else:
+            headers = ["question_id", "text", "grade", "feedback"]
```

Comment:
This is if there're no entries in the array. Since we switched the model to be an array of dictionaries, if there's nothing in the array the header's won't be there.

Reply:
But I'm wondering if having the code assume it won't be empty impacts anything? The unit tests still seem to pass, and in practice our content will definitely have data :man_shrugging: .

The main reason I'd like to remove the conditional if we don't have a good reason to keep it is to DRY things up a bit. If we want to handle the case of empty arrays gracefully, perhaps we should only write CSV files in this method if the array is non-empty instead?

Reply:
Gotcha, ok I'll edit them this way.